src/ranged.cpp

```diff
@@ -1147,32 +1147,39 @@ static int draw_turret_aim( const player &p, WINDOW *w, int line_number, const t
     return line_number;
 }
 
-// TODO: Shunt redundant drawing code elsewhere
-std::vector<tripoint> game::pl_target_ui( target_mode mode, item *relevant, int range, const itype *ammo,
-                                          const target_callback &on_mode_change,
-                                          const target_callback &on_ammo_change )
+std::vector<tripoint> target_handler::target_ui( player pc, const targeting_data &args ) {
```

Comment:
You're passing a whole copy of the character, but probably need only a reference/pointer.